File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.go
func minSubArrayLen(target int, nums []int) int {
    	left, currSum := 0, 0

	minLen := float64(len(nums) + 1)

	for right := 0; right < len(nums); right++ {
		currSum += nums[right]
		for currSum >= target {
			minLen = math.Min(minLen, float64(right-left+1))
			currSum -= nums[left]
			left++
		}

	}
	if minLen == float64(len(nums)+1) {
		return 0
	}

	return int(minLen)
}

File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.py
import math

class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        size = len(nums)
        
        if size == 0:
            return 0
        
        left = 0
        right = 0
        total = 0
        
        result = math.inf
        
        while (right < size):
            total += nums[right]
            right += 1
            while (total >= target):
                result = min(result, right-left)
                total -= nums[left]
                left += 1 # When leaving the loop, total < s, i.e. need next right
        
        if result > size:
            return 0
        else:
            return result